integration_tests/src/bin/benchmark/main.rs

```diff
@@ -0,0 +1,302 @@
+        tokio_rt: None,
+        stun_servers: Vec::new().into(),
+    })
+    .unwrap();
+    let pipeline = Arc::new(Mutex::new(pipeline));
+
+    let mut inputs = Vec::new();
+    for i in 0..bench_config.decoder_count {
+        let input_id = InputId(format!("input_{i}").into());
+        inputs.push(input_id.clone());
+
+        Pipeline::register_input(
+            &pipeline,
+            input_id,
+            RegisterInputOptions {
+                input_options: InputOptions::Mp4(Mp4Options {
+                    should_loop: true,
+                    video_decoder: bench_config.video_decoder,
+                    source: Source::File(bench_config.file_path.clone()),
+                }),
+                queue_options: QueueInputOptions {
+                    offset: Some(Duration::ZERO),
+                    required: true,
+                    buffer_duration: None,
+                },
+            },
+        )
+        .unwrap();
+    }
+
+    let output_id = OutputId("output".into());
+    let receiver = Pipeline::register_encoded_data_output(
+        &pipeline,
+        output_id,
+        RegisterOutputOptions {
+            video: Some(OutputVideoOptions {
+                end_condition: PipelineOutputEndCondition::AnyInput,
+                initial: Component::Tiles(TilesComponent {
+                    id: None,
+                    width: Some(bench_config.output_width as f32),
+                    height: Some(bench_config.output_height as f32),
+                    margin: 2.0,
+                    padding: 0.0,
+                    children: inputs
+                        .into_iter()
+                        .map(|i| {
+                            Component::InputStream(InputStreamComponent {
+                                id: None,
+                                input_id: i,
+                            })
+                        })
+                        .collect(),
+                    transition: None,
+                    vertical_align: VerticalAlign::Center,
+                    horizontal_align: HorizontalAlign::Center,
+                    background_color: RGBAColor(128, 128, 128, 0),
+                    tile_aspect_ratio: (16, 9),
+                }),
+            }),
+
+            audio: None,
+            output_options: EncodedDataOutputOptions {
+                audio: None,
+                video: Some(VideoEncoderOptions::H264(H264OutputOptions {
+                    preset: bench_config.output_encoder_preset,
+                    resolution: Resolution {
+                        width: bench_config.output_width as usize,
+                        height: bench_config.output_height as usize,
+                    },
+                    raw_options: Vec::new(),
+                })),
+            },
+        },
+    )
+    .unwrap();
+
+    Pipeline::start(&pipeline);
+
+    let start_time = Instant::now();
+    while Instant::now() - start_time < bench_config.warm_up_time {
```

Comment:
```suggestion
    while start_time.elapsed() < bench_config.warm_up_time {
```

Reply:
alternative approach would be to just do thread sleep and reading from receiver in a separate channel